### PR TITLE
Feature/ios logging

### PR DESCRIPTION
**Pull request by roesslerj**
We want more logs in regards to the apple responses...

### CODE REVIEW

File: services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/apitoken/ApiTokenService.java [hunk 1]
```diff
@@ -125,9 +127,14 @@ private void updatePerDeviceData(String deviceToken, String transactionId) {
         false,
         false);
     try {
-      iosDeviceApiClient.updatePerDeviceData(jwtProvider.generateJwt(), updateRequest);
+      ResponseEntity<String> response =
+          iosDeviceApiClient.updatePerDeviceData(jwtProvider.generateJwt(), updateRequest);
+      if (response.getStatusCodeValue() != HttpStatus.OK.value() || !"OK".equalsIgnoreCase(response.getBody())) {
```
Comment (FelixRottler) [inline services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/apitoken/ApiTokenService.java:132]: Maybe we can use cwa dev app to trigger a valid update request and see the actual response. Based on this we could craft maybe a more precise check?

File: services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/devicedata/PerDeviceDataValidator.java [hunk 2]
```diff
@@ -60,6 +65,10 @@ public PerDeviceDataResponse validateAndStoreDeviceToken(String transactionId, S
               deviceToken,
               transactionId,
               currentTimeStamp));
+      if (response.getStatusCodeValue() != HttpStatus.OK.value() || !"OK".equalsIgnoreCase(response.getBody())) {
```
Comment (FelixRottler) [inline services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/devicedata/PerDeviceDataValidator.java:68]: Same here

File: services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/apitoken/ApiTokenService.java (hunk 1 above)
Comment (roesslerj) [reply services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/apitoken/ApiTokenService.java]: Done. Thanks for the support :-)

**Review by ioangut — APPROVED** [review]

File: services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/devicedata/PerDeviceDataValidator.java [hunk 3]
```diff
@@ -61,6 +66,10 @@ public PerDeviceDataResponse validateAndStoreDeviceToken(String transactionId, S
               transactionId,
               currentTimeStamp));
       perDeviceDataResponseOptional = parsePerDeviceData(response);
+      if (perDeviceDataResponseOptional.isEmpty()) {
+        logger.warn("Received status {} with body {} and headers {} when trying to update data for iOS device.",
```
Comment (FelixRottler) [inline services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/devicedata/PerDeviceDataValidator.java:70]: This is not for updating but for querying
```suggestion
        logger.warn("Received status {} with body {} and headers {} when trying to query data for iOS device.",
```

File: services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/apitoken/ApiTokenService.java [hunk 4]
```diff
@@ -127,7 +129,8 @@ private void updatePerDeviceData(String deviceToken, String transactionId) {
     try {
       iosDeviceApiClient.updatePerDeviceData(jwtProvider.generateJwt(), updateRequest);
     } catch (FeignException e) {
-      logger.debug("Received Ios API client exception: ", e);
+      logger.warn("Received iOS API client exception with status {} with body {} and headers {}: ",
```
Comment (FelixRottler) [inline services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/apitoken/ApiTokenService.java:132]:
```suggestion
      logger.warn("Received iOS API client exception when updating apple device data with status {} with body {} and headers {}: ",
```

**Review by FelixRottler — CHANGES_REQUESTED** [review]
Change log message when actually querying the device data. Log indicates that this is an update

**Review by FelixRottler — APPROVED** [review]

**Pull request merged**